jvm/src/main/scala/cats/jvm/Task.scala

```diff
@@ -0,0 +1,166 @@
+package cats
+package jvm
+
+import data.Xor
+import scala.reflect.ClassTag
+import java.util.concurrent.{Callable, CountDownLatch, ExecutorService, Executors, ThreadFactory}
+
+/**
+ * A Task is an abstraction of a computation which produces an A
+ * possibly asynchronously
+ */
+sealed trait Task[A] { self =>
+  import Task._
+
+  def map[B](f: A => B): Task[B] = new Task[B] {
+    def eval = self.eval.map(f)
+  }
+
+  def flatMap[B](f: A => Task[B]): Task[B] =
+    new Bind(this, f)
+
+  /**
+   * Returns a new Task which will evaluate the computation in this
+   * Task, but catch any non-fatal exception thrown by the running of
+   * the computation.
+   */
+  def catchNonFatal: Task[Throwable Xor A] =
+    new AttemptNonFatal(this)
+
+  /**
+   * Returns a new Task which will evaluate the computation in this
+   * Task, but catch some subset of the exceptions that might be
+   * thrown by the running of the computation.
+   */
+  def catchOnly[T >: Null <: Throwable: ClassTag]: Task[T Xor A] =
+    new AttemptOnly[T, A](this)
+
+
+  /**
+   * Run the computation to produce an A
+   */
+  final def unsafePerformIO(): A = eval.value
+
+  /**
+   * Run this computation asynchronously, and call the callback with the result
+   */
+  final def unsafeAsyncIO(cb: A => Unit)(implicit pool: ExecutorService): Unit = {
+    val _ = pool.submit(new Callable[Unit] {
+                          def call = cb(Task.this.unsafePerformIO)
+                        })
+  }
+
+  /**
+   * Returns a Task that will produce the same value is this, but
+   * running this resulting task will submit the work to the given
+   * ExecutorService
+   */
+  def fork(implicit pool: ExecutorService): Task[A] = async { cb =>
+    val _ = pool.submit(new Callable[Unit] {
+                                   def call = cb(Task.this.unsafePerformIO)
+                                 })
+  }
+
+  protected def eval: Eval[A]
+}
+
+object Task extends TaskInstances{
+
+  /**
+   * Thread factory to mark all threads as daemon
+   */
+  lazy val DaemonThreadFactory = new ThreadFactory {
+    val defaultThreadFactory = Executors.defaultThreadFactory()
+    def newThread(r: Runnable) = {
+      val t = defaultThreadFactory.newThread(r)
+      t.setDaemon(true)
+      t
+    }
+  }
+
+  /**
+   * The default executor service is a fixed thread pool with N daemon threads,
+   * where N is equal to the twice number of available processors.
+   */
+  def createFixedExecutor(numThreads: Int): ExecutorService = {
+    Executors.newFixedThreadPool(numThreads, DaemonThreadFactory)
```

Comment:
yeah, totally valid point. How about I just avoid providing these and we can let the users just "new ForkJoinPool"